packages/tailwindcss/src/utilities.test.ts

```diff
@@ -15269,3 +15269,681 @@ describe('custom utilities', () => {
+    async function run(candidates: string[]) {
+      let compiled = await compile(
+        css`
+          @plugin "my-plugin";
+
+          @tailwind utilities;
+
+          @theme reference {
+            --breakpoint-lg: 1024px;
+          },
+        `,
+        {
+          async loadPlugin() {
+            return ({ matchUtilities }) => {
+              matchUtilities(
+                {
+                  'border-block': (value, { modifier }) => ({
+                    '--my-modifier': modifier ?? 'none',
+                    'border-block-width': value,
+                  }),
+                },
+                {
+                  values: {
+                    DEFAULT: '1px',
+                    '2': '2px',
+                  },
+
+                  modifiers: {
+                    foo: 'foo',
+                  },
+                },
+              )
+            }
+          },
+        },
+      )
+
+      return compiled.build(candidates)
+    }
+
+    expect(
+      optimizeCss(
+        await run(['border-block', 'border-block-2', 'border-block/foo', 'border-block-2/foo']),
+      ).trim(),
+    ).toMatchInlineSnapshot(`
+      ".border-block {
+        --my-modifier: none;
+        border-block-width: 1px;
+      }
+
+      .border-block-2 {
+        --my-modifier: none;
+        border-block-width: 2px;
+      }
+
+      .border-block-2\\/foo {
+        --my-modifier: foo;
+        border-block-width: 2px;
+      }
+
+      .border-block\\/foo {
+        --my-modifier: foo;
+        border-block-width: 1px;
+      }"
+    `)
+
+    expect(
+      optimizeCss(await run(['border-block/unknown', 'border-block-2/unknown'])).trim(),
+    ).toEqual('')
+  })
+
+  // We're not married to this behavior — if there's a good reason to do this differently in the
+  // future don't be afraid to change what should happen in this scenario.
+  describe('plugins that handle a specific arbitrary value type prevent falling through to other plugins if the result is invalid for that plugin', () => {
+    test('implicit color modifier', async () => {
+      async function run(candidates: string[]) {
+        let compiled = await compile(
+          css`
+            @tailwind utilities;
+            @plugin "my-plugin";
+          `,
+          {
+            async loadPlugin() {
+              return ({ matchUtilities }) => {
+                matchUtilities(
+                  {
+                    scrollbar: (value) => ({ 'scrollbar-color': value }),
+                  },
+                  { type: ['color', 'any'] },
+                )
+
+                matchUtilities(
+                  {
+                    scrollbar: (value) => ({ 'scrollbar-width': value }),
+                  },
+                  { type: ['length'] },
+                )
+              }
+            },
+          },
+        )
+
+        return compiled.build(candidates)
+      }
+
+      expect(optimizeCss(await run(['scrollbar-[2px]/50'])).trim()).toEqual('')
```

Comment:
Should we add a check here that verifies that the fallback actually works?
```suggestion
      expect(optimizeCss(await run(['scrollbar-[2px]', 'scrollbar-[#08c]', 'scrollbar-[#08c]/50'])).trim()).toMatchInlineSnapshot()
      expect(optimizeCss(await run(['scrollbar-[2px]/50'])).trim()).toEqual('')
```

Reply:
Or was the intention to test this in another test: `custom functional utilities with different types`

(Just reading this test on its own is a bit confusing to me)

Reply:
The intent was to make sure that even if `scrollbar-[2px]` works the class `scrollbar-[2px]/50` explicitly generates nothing because it's nonsensical — basically we don't fall through to other plugins in the face of an invalid modifier.

Reply:
Perhaps including both and some comments would help here. I'll make that change in the am.